packages/babel-core/src/config/plugin.js

```diff
@@ -2,8 +2,8 @@
 
 export default class Plugin {
   constructor(plugin: {}, key?: string) {
-    if (plugin.name != null && typeof plugin.name !== "string") {
-      throw new Error("Plugin .name must be a string, null, or undefined");
+    if (!plugin.name || typeof plugin.name !== "string") {
```

Comment:
Hmm, sorry I've been a bit out of the loop. Is there documentation somewhere for why we want to require the name? I'm concerned it'll make it harder for people to upgrade to 7.x